Extension/src/Debugger/configurationProvider.ts

```diff
@@ -231,11 +231,61 @@ class CppConfigurationProvider implements vscode.DebugConfigurationProvider {
                     config.pipeTransport.pipeProgram = replacedPipeProgram;
                 }
             }
+
+            const macOSMIMode: string = config.osx?.MIMode ?? config.MIMode;
+            const macOSMIDebuggerPath: string = config.osx?.miDebuggerPath ?? config.miDebuggerPath;
+
+            // Validate LLDB-MI
+            if (os.platform() === 'darwin' && // Check for macOS
+                (!macOSMIMode || macOSMIMode === 'lldb') &&
+                !macOSMIDebuggerPath && // User did not provide custom lldb-mi
+                !this.hasLLDBFramework()
+                ) {
+
+                const moreInfoMessage: string = localize("lldb.framework.more.info", "More Info");
+                const LLDBFrameworkMissingMessage: string = localize("lldb.framework.not.found", "Unable to locate 'LLDB.framework' for lldb-mi.");
+
+                vscode.window.showErrorMessage(LLDBFrameworkMissingMessage, moreInfoMessage)
+                 .then(value => {
+                    if (value === moreInfoMessage) {
+                        let helpURL: string = "https://aka.ms/vscode-cpptools/LLDBFrameworkNotFound";
+                        vscode.env.openExternal(vscode.Uri.parse(helpURL));
+                    }
+                });
+
+                return undefined;
+            }
         }
         // if config or type is not specified, return null to trigger VS Code to open a configuration file https://github.com/Microsoft/vscode/issues/54213
         return config && config.type ? config : null;
     }
 
+    private hasLLDBFramework(): boolean {
+        const LLDBFramework: string = "LLDB.framework";
+        let searchPaths: string[] = [
+            "/Library/Developer/CommandLineTools/Library/PrivateFrameworks", // XCode CLI
+            "/Applications/Xcode.app/Contents/SharedFrameworks" // App Store XCode
+        ];
+
+        for (const searchPath of searchPaths) {
+            if (fs.existsSync(path.join(searchPath, LLDBFramework))) {
+                return true;
+            }
+        }
+
+        const outputChannel: logger.Logger = logger.getOutputChannelLogger();
+
+        outputChannel.appendLine(localize("lldb.find.failed", "Missing dependency '{0}' for lldb-mi executable.", LLDBFramework));
+        outputChannel.appendLine(localize("lldb.search.paths", "Searched in:"));
+        searchPaths.forEach(searchPath => {
+            outputChannel.appendLine(`\t${searchPath}`);
+        });
+        outputChannel.appendLine(localize("lldb.install.help", "To resolve this issue, either install XCode through the Apple App Store or install the XCode Command Line Tools by running 'xcode-select --install' in a Terminal window."));
```

Comment:
The localizers might be able to determine that 'xcode-select --install' is a command and should not be translated.  To be sure, you could put {0} here and pass in the string for substitution.